src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.cs

```diff
@@ -580,24 +569,35 @@ internal void AssertEquivalentTo(SymbolTreeInfo other)
             var baseTypeNameIndex = BinarySearch(baseTypeName);
             var derivedTypeIndices = _inheritanceMap[baseTypeNameIndex];
 
-            var builder = ArrayBuilder<INamedTypeSymbol>.GetInstance();
-            using var _ = ArrayBuilder<ISymbol>.GetInstance(out var tempBuilder);
-
-            foreach (var derivedTypeIndex in derivedTypeIndices)
+            ArrayBuilder<INamedTypeSymbol>? builder = null;
+            ArrayBuilder<ISymbol>? tempBuilder = null;
```

Comment:
it feels liek thsi would be much simpler and clearer if we just made two builders here unilaterally.  one to actually store the results, and hte other as a temp scratch buffer that could be passed around and used at sublevels, and cleared when done with.

Reply:
This assumes at least one object will be stored in these builders for the majority of calls. According to performance traces this is not the case, resulting in significant overhead for attempting to obtain and return the builders.

Reply:
➡️ Documented the intentional lazy initialization

Reply:
➡️ Transitioned to `TemporaryArray<T>` which does not need this documentation